src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -3090,20 +3087,36 @@ var projectUniqueNamesForBuildIntToUpdate
                     .Distinct()
                     .ToList();
 
-                var successfulFrameworks = allFrameworks
-                    .Except(unsuccessfulFrameworks)
-                    .ToList();
+                var originalFrameworks = updatedPackageSpec
+                    .TargetFrameworks
+                    .ToDictionary(x => x.FrameworkName, x => x.TargetAlias);
```

Comment:
This looks like it'll cause problems when TFMs as aliases gets implemented. Can framework name and target alias be swapped, so the alias is the key, not the value?

Looks like `BuildIntegratedInstallationContext` already exists, not created by this PR, and already uses NuGetFramework as the dictionary key, so my request is out of scope for this PR. 😭

Reply:
yeah it's one of many things that'll need tweaked :)